### ISSUE TITLE
std::bad_alloc errors on the test suite 

### ISSUE THREAD

**Issue by farnoy** [ROCm-Developer-Tools/HIP#823]
cc @sunway513 

Coming here from ROCmSoftwarePlatform/tensorflow-upstream#270, every test in this repo is failing for me with the same error:
[ctest-verbose-output.txt](https://github.com/ROCm-Developer-Tools/HIP/files/2699781/ctest-verbose-output.txt)

Stacktrace from one of the tests:

```       
Message: Process 106909 (hipDeviceMalloc) of user 1000 dumped core.
                
                Stack trace of thread 106909:
                #0  0x00007fa42f357d7f raise (libc.so.6)
                #1  0x00007fa42f342672 abort (libc.so.6)
                #2  0x00007fa4307ee58e _ZN9__gnu_cxx27__verbose_terminate_handlerEv (libstdc++.so.6)
                #3  0x00007fa4307f4dfa _ZN10__cxxabiv111__terminateEPFvvE (libstdc++.so.6)
                #4  0x00007fa4307f4e57 _ZSt9terminatev (libstdc++.so.6)
                #5  0x00007fa4307f50ac __cxa_throw (libstdc++.so.6)
                #6  0x00007fa4307f5647 _Znwm (libstdc++.so.6)
                #7  0x00007fa430838bce _ZN9__gnu_cxx13new_allocatorIcE8allocateEmPKv (libstdc++.so.6)
                #8  0x00007fa430838fa7 _ZNSs9_M_mutateEmmm (libstdc++.so.6)
                #9  0x00007fa43083919c _ZNSs12_M_leak_hardEv (libstdc++.so.6)
                #10 0x00007fa42de795a0 _ZNSt8__detail9_CompilerISt12regex_traitsIcEE22_M_insert_char_matcherILb0ELb0EEEvv (libmcwamp_hsa.so)
                #11 0x00007fa42de768ef _ZNSt8__detail9_CompilerISt12regex_traitsIcEE7_M_atomEv (libmcwamp_hsa.so)
                #12 0x00007fa42de75bd2 _ZNSt8__detail9_CompilerISt12regex_traitsIcEE14_M_alternativeEv (libmcwamp_hsa.so)
                #13 0x00007fa42de73a56 _ZNSt8__detail9_CompilerISt12regex_traitsIcEE14_M_disjunctionEv (libmcwamp_hsa.so)
                #14 0x00007fa42de76d5a _ZNSt8__detail9_CompilerISt12regex_traitsIcEE7_M_atomEv (libmcwamp_hsa.so)
                #15 0x00007fa42de75bd2 _ZNSt8__detail9_CompilerISt12regex_traitsIcEE14_M_alternativeEv (libmcwamp_hsa.so)
                #16 0x00007fa42de73a56 _ZNSt8__detail9_CompilerISt12regex_traitsIcEE14_M_disjunctionEv (libmcwamp_hsa.so)
                #17 0x00007fa42de730cc _ZNSt8__detail9_CompilerISt12regex_traitsIcEEC2EPKcS5_RKSt6localeNSt15regex_constants18syntax_option_typeE (libmcwamp_hsa.so)
                #18 0x00007fa42de72efa _ZNSt11basic_regexIcSt12regex_traitsIcEEC2IPKcEET_S6_NSt15regex_constants18syntax_option_typeE (libmcwamp_hsa.so)
                #19 0x00007fa42de24344 n/a (libmcwamp_hsa.so)
                #20 0x00007fa43095254a call_init.part.0 (ld-linux-x86-64.so.2)
                #21 0x00007fa43095264a _dl_init (ld-linux-x86-64.so.2)
                #22 0x00007fa430956533 dl_open_worker (ld-linux-x86-64.so.2)
                #23 0x00007fa42f457f57 _dl_catch_exception (libc.so.6)
                #24 0x00007fa430955dff _dl_open (ld-linux-x86-64.so.2)
                #25 0x00007fa43076115a n/a (libdl.so.2)
                #26 0x00007fa42f457f57 _dl_catch_exception (libc.so.6)
                #27 0x00007fa42f457ff3 _dl_catch_error (libc.so.6)
                #28 0x00007fa4307618bf n/a (libdl.so.2)
                #29 0x00007fa4307611fa dlopen (libdl.so.2)
                #30 0x00007fa430142e46 _ZN6Kalmar5CLAMP16GetOrInitRuntimeEv (libmcwamp.so)
                #31 0x00007fa430143809 _ZN6Kalmar15KalmarBootstrapC2Ev (libmcwamp.so)
                #32 0x00007fa430143759 __hcc_shared_library_init (libmcwamp.so)
                #33 0x00007fa43095254a call_init.part.0 (ld-linux-x86-64.so.2)
                #34 0x00007fa43095264a _dl_init (ld-linux-x86-64.so.2)
                #35 0x00007fa43094403a _dl_start_user (ld-linux-x86-64.so.2)
```

Full core dump attached  here: [hipDeviceMalloc.zip](https://github.com/ROCm-Developer-Tools/HIP/files/2699787/hipDeviceMalloc.zip)

**Comment by farnoy** [comment]
```sh
$ grep -R . /opt/rocm/.info/                                                                                                                                
/opt/rocm/.info/version-dev:2.0.89
/opt/rocm/.info/version-libs:2.0.89
/opt/rocm/.info/version-utils:2.0.89
```

**Comment by farnoy** [comment]
So I recompiled hcc completely and replaced the `/opt/rocm/hcc` directory entirely, which makes HIP tests pass except for these two:

```
	 96 - directed_tests/runtimeApi/memory/hipMemset3D-N10--memsetval0x42.tst (Child aborted)
	108 - directed_tests/runtimeApi/stream/hipStreamCreateWithPriority.tst (Failed)
```

Testing with tensorflow, the stacktrace changes and crashed in `libhip_hcc`. When I replaced `/opt/rocm/hip` with my `hip` build, the stacktrace changes again and crashes in `libhiprand.so`.

I guess there must be some incompatibility in libstdc++ between Ubuntu and ArchLinux, but I'm not sure. I'll try recompiling more packages and see if that helps.
